docs/usage.md

```diff
@@ -0,0 +1,151 @@
+
+### Instance MCF Generation Mode (`genmcf`)
+
+To run the tool in genmcf mode, use:
+  ```bash
+  dc-import genmcf <list of csv/tmcf files>
+  ```
+
+Optionally, schema file(s) may also be passed. This is required to resolve references to newly introduced schema nodes.
+
+For example, we can use `genmcf` to perform validations, and generate instance MCF from test files about COVID-19 cases in India.
+
+These test files are:
+- [covid.csv](../tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.csv) at path `tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.csv` relative to the base of this repo.
+- [covid.tmcf](../tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.tmcf) at path `tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.tmcf` relative to the base of this repo.
+
+From the base of the repo, we issue the following command:
+  ```bash
+  dc-import lint tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.csv tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/input/covid.tmcf
+  ```
+
+This will output issues found in the input to [`dc_generated/report.json`](../tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/output/report.json) and [`dc_generated/summary_report.html`](../tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/output/summary_report.html) under the current working directory.
+
+This will also output the instance MCFs generated from the template to [`dc_generated/table_mcf_nodes_covid.mcf`](../tool/src/test/resources/org/datacommons/tool/genmcf/statchecks/output/table_mcf_nodes_covid.mcf). Note that instance MCF will not be generated if there are any fatal errors in the input files. These fatal errors will instead be logged to `report.json` and `summary_report.html`.
+
+## Flags
+
+Flags available for the tool are listed below. All flags apply to both usage modes (`lint` and `genmcf`).
+
+You can also run `dc-import --help` to see a list of flags in your terminal.
+
+### `-e`, `--existence-checks`
+
+Checks DCID references to schema nodes against the KG and locally. If this flag is set, then calls will be made to the Staging API server, and instance MCFs get fully loaded into memory. 
+
+Defaults to `true`.
+
+### `-h`, `--help`
+
+Shows a help message and exit.
+
+### `-n`, `--num-threads=<numThreads>`
+
+Specifies the number of concurrent threads used for processing CSVs.
+
+Defaults to `1`.
+
+### `-o`, `--output-dir=<outputDir>`
+
+Specifies the directory to write output files.
+
+Default is `dc_generated/` within current working directory.
+
+### `-s`, `--stat-checks`
+
+Checks integrity of time series by checking for holes, variance in values, etc.
```

Comment:
This results in a set of counters (e.g., ...) reported in `report.json`.  For every such counter, there is a few exemplar cases to help the user understand and resolve the issue.